### PR TITLE
Add contributor guidelines

### PR DESCRIPTION
**Pull request by Eleonore9**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,11 +30,51 @@ Current status:
 
 See the [docs](https://github.com/MastodonC/witan.models.demography/blob/master/doc/intro.md) for more info about the methodology.
 
-## Development tools
+
+## Contribute to `witan.models.demography`
+
+### General practices
+
+If you wish to contribute to `witan.models.demography`, please follow read the following guidelines.
```
Comment (sunnytownsend) [inline README.md]: delete 'follow'

File: README.md [hunk 2]
```diff
@@ -15,11 +30,51 @@ Current status:
...
+
+#### Github guidelines
+* **Fork this repository** (or clone it if you have writing writes to the reposotory).
```
Comment (sunnytownsend) [inline README.md]: typo: writing rights to the repository

File: README.md [hunk 3]
```diff
@@ -15,11 +30,51 @@ Current status:
...
+* **Create a new branch**. Let's try and keep the following naming conventions for branches:
+  * `feature/<what-this-branch-solves>`, example: `feature/exception-logging`
```
Comment (sunnytownsend) [inline README.md]: Its a small thing, and feel free to not do it, but could you make this a model type of feature? e.g. 'feature/project-mortality-rates' or something

File: README.md [hunk 4]
```diff
@@ -15,11 +30,51 @@ Current status:
...
+  * `doc/<what-this-branch-solves>`, example: `doc/contributors-practices`
+  * `fix/<what-this-branch-solves>`, example: `fix/run-ccm-from-cli`
```
Comment (sunnytownsend) [inline README.md:45]: interesting that the fix branch name is the same as the feature branch name - that makes sense though!

File: README.md [hunk 5]
```diff
@@ -15,11 +30,51 @@ Current status:
...
+  * `tidy-up/<what-this-branch-solves>`, example: `tidy-up/upgrade-deps`
+
+  This way, when you see a branch starting by `fix/` we know something is broken and someone is repairing it.
+* **Keep branches short** so that the reviewing process is easier and faster
+* Start a pull request (PR) as earlier as possible. You can add a `WIP` in the title to specify it's in progress.
```
Comment (sunnytownsend) [inline README.md]: typo: early

File: README.md (hunk 4 above)
Comment (Eleonore9) [inline README.md:45]: @sunnytownsend what do you mean by the branches names are the same?

**Comment by sunnytownsend** [comment]
Do you know if we have had any contributors yet?

File: README.md [hunk 6]
```diff
@@ -15,11 +30,51 @@ Current status:
...
+  * **run the lint tools** [`Eastwood`](https://github.com/jonase/eastwood) and [`Kibit`](https://github.com/jonase/kibit) with `$ lein eastwood` and `$ lein kibit`.
+  * **format your code** with [`Cljfmt`](https://github.com/weavejester/cljfmt) tool with `lein cljfmt check` followed by `lein cljfmt fix`.
+  * **check the cohort-component model can still be run** from the command-line.
+  You'll need to build the jar file, run it on your terminal and check it outputs population projections.
+  For that follow the instructions in the ["user section"](#use-witanmodelsdemography) of the Readme.
+
+#### Coding guidelines
+
+* Write unit tests, docstrings, and documentation.
+* As much as possible try to not have data changes and code changes on the same branch or the same commit.
```
Comment (sunnytownsend) [inline README.md]: maybe be clearer to say 'Try not to have data changes and code changes in the same commit, and preferably not the same branch, as the data tends to swamp the code and hinder reviewing'.

**Comment by Eleonore9** [comment]
@sunnytownsend nope, no contributer outside of MC or GLA. The repo hasn't been forked and nobody submitted a PR so far. 

In terms of users there is an early version of the project on Clojars. Jarkeeper says it's been downloaded 34 times but I don't know what that actually means: https://jarkeeper.com/MastodonC/witan.models.demography

For now the "contributors guidelines" are for us, but they could be used by an external person in the future.

File: README.md [hunk 7]
```diff
@@ -122,13 +182,13 @@ The geographical area for the projection must be specified. This is done with th
   * a London borough (starts with "E09")
   * Northern Ireland, Scotland, or Wales (country codes "N92000002", "S92000003", or "W92000004"; smaller geographies not yet available)
```
Comment (sunnytownsend) [inline README.md]: can you remove this please Eleonore, as will be removing from the data too

**Comment by sunnytownsend** [comment]
:shipit: once last comment addressed

**Pull request merged**